packages/react-devtools-shared/src/devtools/views/SuspenseTab/SuspenseTimeline.js

```diff
@@ -58,19 +67,28 @@ function getSuspendableDocumentOrderSuspense(
     }
   }
 
-  return suspenseTreeList;
+  return list;
 }
 
 function SuspenseTimelineInput({rootID}: {rootID: Element['id'] | void}) {
   const bridge = useContext(BridgeContext);
-  const store = useContext(StoreContext);
+  const store = useSuspenseStore();
   const dispatch = useContext(TreeDispatcherContext);
   const {highlightHostInstance, clearHighlightHostInstance} =
     useHighlightHostInstance();
+  const [uniqueSuspendersOnly, setUniqueSuspendersOnly] = useState(true);
+  function handleToggleUniqueSuspenders(event: SyntheticEvent) {
+    // TODO: Currently loses timeline position because position is timeline-index-based not suspense-id-based.
```

Comment:
Requires a larger refactor. The timeline should just live in state like the Components tree related things. I noticed some more bugs with the backend reconciler though so I'll focus on that.